### ISSUE TITLE
No support for retina/high-DPI displays

### ISSUE THREAD

**Issue by j-f1** [ankitrohatgi/WebPlotDigitizer#276]
Currently, users with a Retina/HiDPI display see a pixellated image, even if the actual image is much higher resolution. It would be great to check the `devicePixelRatio` property and multiply visible canvas resolutions by that value to make things render pixel-perfect ✨

I tried to do this (using [this method](https://github.com/observablehq/stdlib/blob/main/src/dom/context2d.js) to scale the user visible  canvas contexts appropriately), but it didn’t seem to work very well, probably because of how the canvas content is copied from another canvas. I’m not sure how deep the changes need to go for this to work right.

**Comment by billdenney** [comment]
Is this somewhat similar to #176?  Specifically, is the source you're working from some form of vector graphic like a .pdf?

**Comment by j-f1** [comment]
No, my source is a (somewhat low-resolution) raster source, but the actual UI elements (like the dots and such added by WPD) are what I’m talking about here — as well as the image itself when zoomed out.
